src/main/java/com/flightstats/hub/dao/aws/MaxItemsEnforcerResource.java

```diff
@@ -0,0 +1,40 @@
+package com.flightstats.hub.dao.aws;
+
+import com.flightstats.hub.rest.PATCH;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.inject.Inject;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+
+import java.util.Collections;
+
+import static javax.ws.rs.core.Response.ok;
+
+@Slf4j
+@Path("/internal/max-items")
+public class MaxItemsEnforcerResource {
+
+    private final S3Config s3Config;
+
+    @Inject
+    public MaxItemsEnforcerResource(S3Config s3Config) {
+        this.s3Config = s3Config;
+    }
+
+    @PATCH
+    @Path("/{channel}")
+    public Response enforce(@PathParam("channel") String channel) {
+        try {
+            S3Config.S3ConfigLockable s3ConfigLockable = s3Config.new S3ConfigLockable(Collections.emptyList());
```

Comment:
I usually try to avoid calling non-static nested classes from outside the class it's defined in...it breaks encapsulation, gets confusing, and tends to indicate that responsibilities are a bit muddled.